src/precice/impl/ParticipantImpl.cpp

```diff
@@ -550,7 +550,13 @@ double ParticipantImpl::getMaxTimeStepSize() const
 {
   PRECICE_CHECK(_state != State::Finalized, "getMaxTimeStepSize() cannot be called after finalize().");
   PRECICE_CHECK(_state == State::Initialized, "initialize() has to be called before getMaxTimeStepSize() can be evaluated.");
-  return _couplingScheme->getNextTimeStepMaxSize();
+  const double nextTimeStepSize = _couplingScheme->getNextTimeStepMaxSize();
+  PRECICE_ASSERT(!math::equals(nextTimeStepSize, 0.0));
+  PRECICE_ASSERT(math::greater(nextTimeStepSize, 0.0));
+  if (not math::greater(nextTimeStepSize, 0.0, 100 * math::NUMERICAL_ZERO_DIFFERENCE)) {
+    PRECICE_WARN("preCICE just returned a maximum time step size of {}. This can happen, if you are using many substeps per time window over multiple time windows due to added-up differences of machine precision. Please check your criterion for setting the time step size in your adapter.", nextTimeStepSize);
```

Comment:
I removed the `bug` tag, because this issue is a usability problem only. With the current implementation the user is responsible for picking the time step size with the correct strategy. If we want, we can still raise the warning above. I think such a small time step size is still worth a warning.